gcc/testsuite/rust/compile/torture/union.rs

```diff
@@ -0,0 +1,35 @@
+// { dg-do compile }
+// { dg-options "-w" }
```

Comment:
what does -w do here i wonder

Reply:
`       -w  Inhibit all warning messages.`
I guess it's a trick to avoid the test from failing on warnings

Reply:
Are there really so many diagnostics to make it infeasible to spell these out individually?  The problem that I have with `-w` and similar "hammers" is that also you won't notice any bogus diagnostics appearing in the future.

Reply:
Yeah thats a good point I think @thomasyonug is fixing up our unused variables at the moment which will let us remove all of the duplicate warnings we have at the moment. Rustc as far as I know controls warnings via the attribute system but I am wondering if a flag system might also prove useful.

Reply:
Aha, per <http://mid.mail-archive.com/YQciMeKNpCH+ZMsJ@wildebeest.org> the `-w` here is just an interim thing.

Reply:
May I should add this new commit here then...